pkg/controller/periodic/periodic.go

```diff
@@ -6,42 +6,58 @@ import (
 	"sort"
 	"time"
 
-	"k8s.io/klog/v2"
-
 	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
 	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/klog/v2"
 
-	"github.com/openshift/insights-operator/pkg/config"
+	"github.com/openshift/insights-operator/pkg/anonymization"
+	"github.com/openshift/insights-operator/pkg/config/configobserver"
 	"github.com/openshift/insights-operator/pkg/controller/status"
 	"github.com/openshift/insights-operator/pkg/controllerstatus"
 	"github.com/openshift/insights-operator/pkg/gather"
 	"github.com/openshift/insights-operator/pkg/recorder"
 )
 
-type Configurator interface {
-	Config() *config.Controller
-	ConfigChanged() (<-chan struct{}, func())
-}
-
+// Controller periodically runs gatherers, records their results to the recorder
+// and flushes the recorder to create archives
 type Controller struct {
-	configurator Configurator
-	recorder     recorder.FlushInterface
-	gatherers    map[string]gather.Interface
-	statuses     map[string]*controllerstatus.Simple
+	configurator                 configobserver.Configurator
+	recorder                     recorder.FlushInterface
+	gatherers                    []gather.Interface
+	gatherersLastProcessingTimes map[string]time.Time
```

Comment:
Do we need this map? Sorry, maybe I wasn't eplicit in my previous review. My thinking was roughly as follows:
- add the new time attribute to the `Gatherer` struct
- access the time by the "getter" method via the `Interface` (as you already updated it)
In other words it's about merging `Interface` and `LowPriorityGatherer` (btw this is not a good name IMO) into one. What do you think? 
I guess it would require also creating a setter method (to set the last processing time), but you will know better.

Reply:
But then we'd need to repeat this logic in each gatherer. Maybe not a big deal and we can add a few more methods to LowPriorityGatherer, but merging Interface and LowPriorityGatherer into one doesn't feel right to me, cuz then we'd have some useless code in the gatherers that aren't special. 

> btw this is not a good name IMO

Any better ideas?

Reply:
Which logic exactly? I was thinking just about the accessing the new attribute. Hmm yeah my thinking was to have `GetLastProcessedTime` and comparing it to `GetDesiredGatheringPeriod`.... but it's probably true that it's useless for the `clusterconfig` gatherer.

Reply:
Maybe `CustomPeriodGatherer` or something like that?

Reply:
Implemented a little different thing where custom period gatherers just return true or false if they want to be processed and they are responsible for tracking time. It's little easier to unit test. https://github.com/openshift/insights-operator/pull/399/files#diff-095e9955b4ce960ffb4ea94ee251d09818bc2d6a76b8f4b3c5414ee419ed82f1R42 WDYT?

Reply:
@Sergey1011010 Yes I like it. Thank you.